tests/queries/0_stateless/03282_json_equal_comparison.sql

```diff
@@ -0,0 +1,22 @@
+set enable_json_type=1;
+create table test (json1 JSON(max_dynamic_paths=2, a UInt32), json2 JSON(max_dynamic_paths=2, a UInt32)) engine=Memory;
+insert into test format JSONEachRow
+{"json1" : {}, "json2" : {}}
+{"json1" : {"a" : 42}, "json2" : {"a" : 42}}
+{"json1" : {"a" : 42}, "json2" : {"a" : 43}}
+{"json1" : {"a" : 42, "b" : 42}, "json2" : {"a" : 42, "b" : 42}}
+{"json1" : {"a" : 42, "b" : 42}, "json2" : {"a" : 42, "b" : 43}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 43, "c" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 43}}
+{"json1" : {"a" : 42, "b" : 42, "c" : null, "d" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : null}}
+{"json1" : {"a" : 42, "b" : 42, "c" : null, "d" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : null}, "json2" : {"a" : 42, "b" : 42, "d" : null, "c" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : null}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42, "e" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42, "e" : 42}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42, "e" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42, "e" : "42"}}
+{"json1" : {"a" : 42, "b" : 42, "c" : 42, "d" : 42, "e" : 42}, "json2" : {"a" : 42, "b" : 42, "d" : 42, "c" : 42, "e" : 42.0}};
+
+select json1, json2, json1 == json2 from test;
```

Comment:
Can we compare for inequality?

Reply:
Also it's worth adding cases with other operators (expecting error or whatever)

Reply:
> Can we compare for inequality?

Yes, I will add tests for it as well.

Reply:
Added test for inequality.

> Also it's worth adding cases with other operators (expecting error or whatever)

I will make a separate PR with tests when I add an error for other operators as a backward incompatible change under a setting

Reply:
https://github.com/ClickHouse/ClickHouse/pull/73276